generator/internal/descriptor_utils.cc

```diff
@@ -359,6 +332,43 @@ std::string FormatProtobufRequestParameters(
 
 }  // namespace
 
+absl::optional<ResourceRoutingInfo> ParseResourceRoutingHeader(
+    google::protobuf::MethodDescriptor const& method) {
+  if (!method.options().HasExtension(google::api::http)) return {};
+  google::api::HttpRule http_rule =
+      method.options().GetExtension(google::api::http);
+
+  std::string url_pattern;
+  switch (http_rule.pattern_case()) {
+    case google::api::HttpRule::kGet:
+      url_pattern = http_rule.get();
+      break;
+    case google::api::HttpRule::kPut:
+      url_pattern = http_rule.put();
+      break;
+    case google::api::HttpRule::kPost:
+      url_pattern = http_rule.post();
+      break;
+    case google::api::HttpRule::kDelete:
+      url_pattern = http_rule.delete_();
+      break;
+    case google::api::HttpRule::kPatch:
+      url_pattern = http_rule.patch();
+      break;
+    default:
+      GCP_LOG(FATAL) << __FILE__ << ":" << __LINE__
+                     << ": google::api::HttpRule not handled" << std::endl;
+      std::exit(1);
```

Comment:
Does `GCP_LOG(FATAL)` return?  We seem to assume that it does not in a few places, including at the beginning of `DefaultIdempotencyFromHttpOperation()`, just a few lines up.

Reply:
FWIW, it does return, probably should not.